src/Controller/Api/v1/DepartmentController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Controller\Api\v1;
+
+use App\Services\DepartmentService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+#[Route('/api/v1/department', name: 'app_api_v1_department_')]
+class DepartmentController extends AbstractController
+{
+    #[Route('/list', name: 'list', methods: ['GET'])]
+    public function index(DepartmentService $departmentService): JsonResponse
+    {
+        $result = $departmentService->list();
+        if ($result['status']) {
+            return $this->json($result, Response::HTTP_OK);
+        }
+        return $this->json($result, Response::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    #[Route('/create', name: 'create', methods: ['POST'])]
+    public function store(Request $request, DepartmentService $departmentService): JsonResponse
+    {
+        $requestData = [
+            'title' => $request->request->get('title'),
+            'slug' => $request->request->get('slug'),
+            'description' => $request->request->get('description'),
+        ];
+
+        $result = $departmentService->createDepartment($requestData);
+
+        if ($result['status']) {
+            return $this->json($result, Response::HTTP_CREATED);
+        }
+
+        return $this->json($result, Response::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    #[Route('/details/{id}', name: 'show', methods: ['GET'])]
+    public function show(DepartmentService $departmentService, int $id): JsonResponse
+    {
+        $result = $departmentService->details($id);
+        if ($result['status']) {
+            return $this->json($result, Response::HTTP_OK);
+        }
+
+        return $this->json($result, Response::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    #[Route('/update/{id}', name: 'update', methods: ['PUT', 'PATCH', 'POST'])]
+    public function update(Request $request, DepartmentService $departmentService, int $id): JsonResponse
+    {
+        $requestData = [
+            'title' => $request->request->get('title'),
```

Comment:
1. Can't we use $request->get('title') directly instead of $request->request->get('title')? 
2. Why we need to prepare an array here? Can't we pass this $request directly to service?